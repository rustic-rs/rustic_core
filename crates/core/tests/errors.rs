use rstest::{fixture, rstest};

use rustic_core::{ErrorKind, RusticError, Severity, Status};

#[fixture]
fn error() -> Box<RusticError> {
    RusticError::with_source(
        ErrorKind::InputOutput,
        "A file could not be read, make sure the file at `{path}` is existing and readable by the system.",
        std::io::Error::new(std::io::ErrorKind::ConnectionReset, "Networking Error"),
    )
    .attach_context("path", "/path/to/file")
    .attach_context("called", "used s3 backend")
    .attach_status(Status::Permanent)
    .attach_severity(Severity::Error)
    .attach_error_code("C001")
    .append_guidance_line("Appended guidance line")
    .prepend_guidance_line("Prepended guidance line")
    .attach_existing_issue_url("https://github.com/rustic-rs/rustic_core/issues/209")
    .ask_report()
}

#[rstest]
fn test_error_display(error: Box<RusticError>) {
    insta::assert_snapshot!(error);
}

#[rstest]
fn test_error_debug(error: Box<RusticError>) {
    insta::assert_debug_snapshot!(error);
}
