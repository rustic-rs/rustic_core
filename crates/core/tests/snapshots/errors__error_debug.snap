---
source: crates/core/tests/errors.rs
expression: error
---
RusticError {
    kind: InputOutput,
    guidance: "Prepended guidance line\nA file could not be read, make sure the file at `{path}` is existing and readable by the system.\nAppended guidance line",
    docs_url: None,
    error_code: Some(
        "C001",
    ),
    ask_report: true,
    existing_issue_urls: [
        "https://github.com/rustic-rs/rustic_core/issues/209",
    ],
    new_issue_url: None,
    context: [
        (
            "path",
            "/path/to/file",
        ),
        (
            "called",
            "used s3 backend",
        ),
    ],
    source: Some(
        Custom {
            kind: ConnectionReset,
            error: "Networking Error",
        },
    ),
    severity: Some(
        Error,
    ),
    status: Some(
        Permanent,
    ),
    backtrace: Some(
        <disabled>,
    ),
}
